src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoId, setTodoId] = useState(0);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const filteredTodos = useMemo(
+    () => todos.filter(todo => {
+      const filteredTodo = todo.title.toLowerCase().includes(searchQuery.trim().toLowerCase());
```

Comment:
this variable doesnt contain todo
it contains boolean value
```suggestion
      const isTitleFilterMatch  = todo.title.toLowerCase().includes(searchQuery.trim().toLowerCase());
```